toolchains/gcc/defs.bzl

```diff
@@ -60,6 +60,12 @@ GccDistributionInfo = provider(
     },
 )
 
+GdbToolInfo = provider(
+    fields = {
+        "gdb": provider_field(typing.Any),
+    },
+)
```

Comment:
I don't think you need to do this. Instead, just add gdb to the `binary_utilities_info` on line 288 and then have your `toolchain` on line 155 of `openocd/defs.bzl` take a `CxxToolchainInfo`. I think that should work?